mmv1/products/sql/terraform.yaml

```diff
@@ -36,12 +36,25 @@ overrides: !ruby/object:Overrides::ResourceOverrides
           deletion_protection: "false"
         oics_vars_overrides:
           deletion_protection: "false"
+      - !ruby/object:Provider::Terraform::Examples
+        name: "sql_database_deletion_policy"
+        primary_resource_id: "database_deletion_policy"
+        vars:
+          database_name: "my-database"
+          database_instance_name: "my-database-instance"
+          deletion_protection: "true"
+        test_vars_overrides:
+          deletion_protection: "false"
+        oics_vars_overrides:
+          deletion_protection: "false"
     properties:
       collation: !ruby/object:Overrides::Terraform::PropertyOverride
         default_from_api: true
       charset: !ruby/object:Overrides::Terraform::PropertyOverride
         default_from_api: true
         diff_suppress_func: 'caseDiffSuppress'
+    custom_code: !ruby/object:Provider::Terraform::CustomCode
+      custom_delete: templates/terraform/custom_delete/sql_database_deletion_policy.erb
```

Comment:
Can we add this as a pre_delete instead of an entire custom_delete? It should be the top block checking the deletionPolicy and returning and it will prevent us from hand-writing the rest of the delete code

Reply:
Done.